File: ios/FHIRmHealth/HealthKitQueryController.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(HealthKitQueryController, NSObject)

RCT_EXTERN_METHOD(start:(RCTPromiseResolveBlock)resolve
                  reject:(RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(stop:(RCTPromiseResolveBlock)resolve
                  reject:(RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(reset:(RCTPromiseResolveBlock)resolve
                  reject:(RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(status:(RCTPromiseResolveBlock)resolve
                  reject:(RCTPromiseRejectBlock)reject)

+(BOOL)requiresMainQueueSetup {
  return NO;
}

@end

File: ios/FHIRmHealth/HealthKItQueryController.swift
@objc(HealthKitQueryController)
class HealthKitQueryController: NSObject {
  @objc func start(_ resolve: RCTPromiseResolveBlock, reject: RCTPromiseRejectBlock) {
    HealthKitConnector.shared.launchBackgroundQuery()
    resolve(nil)
  }

  @objc func stop(_ resolve: RCTPromiseResolveBlock, reject: RCTPromiseRejectBlock) {
    HealthKitConnector.shared.stopBackgroundQuery()
    resolve(nil)
  }

  @objc func reset(_ resolve: RCTPromiseResolveBlock, reject: RCTPromiseRejectBlock) {
    HealthKitConnector.shared.resetBackgroundQuery()
    resolve(nil)
  }

  @objc func status(_ resolve: RCTPromiseResolveBlock, reject: RCTPromiseRejectBlock) {
    resolve(HealthKitConnector.shared.queryStatus.rawValue)
  }
}
